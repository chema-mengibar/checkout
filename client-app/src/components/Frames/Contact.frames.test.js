import React from "react";
import renderer, { create, act } from "react-test-renderer";
import ContactFrame from "./Contact.frames";

describe("Component Snapshots", () => {
  test("", () => {
    const component = renderer.create(
      <ContactFrame />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Send Button", () => {
  test("should not be disable if the inputs are valid", () => {

    const onSubmitFnMock = jest.fn();
    const component = renderer.create( <ContactFrame fieldsData={null} clicked={ onSubmitFnMock } />  );
    const _root = component.root;
    
    const button = _root.find( element => element.type === 'button');
    const inputName = _root.find( element => element.type === 'input' && element.props.id == 'name');
    
    const event = {
      currentTarget: { value: 'Musterman' }
    };
    act(() => {
      inputName.props.onChange( event );
    });
    expect(button.props.disabled).toEqual(false);
      
  });
});

describe("Component State", () => {
  test(" should have the input 'Name' value", () => {
    
    const onSubmitFnMock = jest.fn();
    const component = renderer.create( <ContactFrame fieldsData={null} clicked={ onSubmitFnMock } />  );
    const _root = component.root;
    
    const button = _root.find( element => element.type === 'button');
    const inputName = _root.find( element => element.type === 'input' && element.props.id == 'name');
    
    const mockGivedName = 'Musterman';

    const event = {
      currentTarget: { value: mockGivedName }
    };
        
    act(() => {
      inputName.props.onChange( event );
    });
    act(() => {
      button.props.onClick( );
    });
    
    const expectedObj = {
      name: mockGivedName,
      prename: '',
      email: '',
      address: '',
      place: '',
      plz: ''
    }
    expect(onSubmitFnMock).toBeCalledWith( expectedObj );

  });
});