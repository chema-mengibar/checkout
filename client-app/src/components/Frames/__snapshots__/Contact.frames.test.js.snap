// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshots  1`] = `
<div
  className="Contactframes__Frame-sc-1sm65g0-0 bIMFVL"
>
  <div
    className="Contactframes__FrameBox-sc-1sm65g0-1 jephfn"
  >
    <div
      className="frame-title"
    >
      Kontakdaten
    </div>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="Contactframes__StyledLabel-sc-1sm65g0-3 kzeYqr form-label"
          htmlFor="name"
        >
          Name
        </label>
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="Contactframes__StyledLabel-sc-1sm65g0-3 kzeYqr form-label"
          htmlFor="vorname"
        >
          Vorname
        </label>
        <input
          className="form-control"
          id="vorname"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="Contactframes__StyledLabel-sc-1sm65g0-3 kzeYqr form-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            }
          }
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="Contactframes__StyledLabel-sc-1sm65g0-3 kzeYqr form-label"
          htmlFor="strasse"
        >
          Stra√üe und Hausnummer
        </label>
        <input
          className="form-control"
          id="strasse"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group--50 form-group"
      >
        <label
          className="Contactframes__StyledLabel-sc-1sm65g0-3 kzeYqr form-label"
          htmlFor="ort"
        >
          Ort
        </label>
        <input
          className="form-control"
          id="ort"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group--50 form-group"
      >
        <label
          className="Contactframes__StyledLabel-sc-1sm65g0-3 kzeYqr form-label"
          htmlFor="plz"
        >
          PLZ
        </label>
        <input
          className="form-control"
          id="plz"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            }
          }
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
  <button
    className="Contactframes__StyledButton-sc-1sm65g0-2 dmwCyj btn btn-primary"
    disabled={true}
    id="contact-button"
    onClick={[Function]}
    type="button"
  >
    Weiter
  </button>
</div>
`;
